src/components/Comments/CommentsBarContext.tsx

```diff
@@ -0,0 +1,93 @@
+import { useCookieState } from 'ahooks';
+import { noop } from 'lodash';
+import {
+  createContext,
+  useCallback,
+  useContext,
+  useMemo,
+  useState,
+} from 'react';
+import { ChildrenProp } from '~/common';
+
+const initialCommentsBarContext = {
+  toggleThreadComments: noop,
+  toggleCommentsBar: noop,
+  isCommentsBarOpen: false,
+  expandedThreads: [] as string[],
+  resourceCommentsTotal: 0,
+  setResourceCommentsTotal: noop,
+};
+
+export const CommentsBarContext = createContext(initialCommentsBarContext);
+
+CommentsBarContext.displayName = 'CommentsBarContext';
+
+export const CommentsBarProvider = ({ children }: ChildrenProp) => {
+  const [cookieCommentsBarOpen, setCookieCommentsBarOpen] = useCookieState(
+    'show-comments',
+    { defaultValue: 'false' }
+  );
+  const [isCommentsBarOpen, setCommentsBarOpen] = useState(
+    cookieCommentsBarOpen === 'true'
+  );
```

Comment:
I think this just duplicates state from the cookie hook.
```suggestion
  const isCommentsBarOpen = cookieCommentsBarOpen === 'true';
```